### ISSUE TITLE
[Bug] Label width not adjusted on text changed

### ISSUE THREAD

**Issue by leechenhwa** [dotnet/maui#1535]
### Description
Label width not updated on text changed

### Steps to Reproduce
First disable XAML Hot Reload in settings of Visual Studio 2019 Preview

1.  Create a MAUI app as MauiApp1 
2.  Deploy to Android Local Device 
3.  In the phone, Touch button CLICK ME to increase the count

### Expected Behavior

When click up to 10 or more, the digital number charaters should be all visible.

### Actual Behavior

Only the first digital number charater is displayed, the right ones seem to be cropped.

### Basic Information

- Version with issue:  MAUI 6.0.100-preview.5.794

- IDE: Visual Studio 2019 Preview
- Platform Target Frameworks: 
  - Android: 30

### Code
```
MauiApp1\MainPage.xaml

<Label Text="Current count: 0  "
                    Grid.Row="2"
                    SemanticProperties.Hint="Counts the number of times you click"
                    FontSize="18"
                    FontAttributes="Bold"
                    x:Name="CounterLabel"
                    HorizontalOptions="Center" />
					
MauiApp1\MainPage.xaml.cs			
int count = 0;
		private void OnCounterClicked(object sender, EventArgs e)
		{
			count++;
			CounterLabel.Text = $"Current count: {count}";
		}					
		
```

With initial text appended with a whitespace char to make it wider for testing.

### Screenshots
The char 0 is displayed partially.

![Screenshot_20210703-142226_crop](https://user-images.githubusercontent.com/16028379/124346872-e89c9f80-dc13-11eb-8980-d0a52ca0458a.png)

**Comment by imgen** [comment]
@leechenhwa this is a duplicate of my bug, :)
https://github.com/dotnet/maui/issues/1452
Judging by your github name, I assume you are from China. I am Chinese too, glad to meet another Chinese on github, :)

**Comment by hartez** [comment]
Fixed by #1700.
